File: Assignments/2_Operators and Expression/Javascript/Solutions/Q10.js
function factorial(n) {
    return n <= 1 ? 1 : n * factorial(n - 1);
}

function toRadians(degrees) {
    return degrees * Math.PI / 180;
}

function sin(x) {
    x = toRadians(x);
    return x - Math.pow(x, 3) / factorial(3) + Math.pow(x, 5) / factorial(5) - Math.pow(x, 7) / factorial(7);
}

function cos(x) {
    x = toRadians(x);
    return 1 - Math.pow(x, 2) / factorial(2) + Math.pow(x, 4) / factorial(4) - Math.pow(x, 6) / factorial(6);
}

const angle = 30;
console.log(`sin(${angle}) = ${sin(angle)}`);
console.log(`cos(${angle}) = ${cos(angle)}`);

File: Assignments/2_Operators and Expression/C#/Q10.cs
using System;

class Program
{
    static void Main()
    {
        Console.Write("Enter the angle in degrees: ");
        double degrees = double.Parse(Console.ReadLine());

        double radians = degrees * Math.PI / 180;

        double sin = CalculateSin(radians);
        double cos = CalculateCos(radians);

        Console.WriteLine($"sin({degrees}) = {sin}");
        Console.WriteLine($"cos({degrees}) = {cos}");
    }

    static double CalculateSin(double x)
    {
        return x - Math.Pow(x, 3) / Factorial(3) + Math.Pow(x, 5) / Factorial(5) - Math.Pow(x, 7) / Factorial(7);
    }

    static double CalculateCos(double x)
    {
        return 1 - Math.Pow(x, 2) / Factorial(2) + Math.Pow(x, 4) / Factorial(4) - Math.Pow(x, 6) / Factorial(6);
    }

    static double Factorial(int n)
    {
        if (n <= 1)
            return 1;
        return n * Factorial(n - 1);
    }
}
